src/gluon/events/member/TeamMemberCreated.ts

```diff
@@ -24,11 +25,16 @@ subscription TeamMemberCreatedEvent {
   }
 }
 `)
-export class TeamMemberCreated implements HandleEvent<any> {
+export class TeamMemberCreated extends BaseQMEvent implements HandleEvent<any> {
 
     public async handle(event: EventFired<any>, ctx: HandlerContext): Promise<HandlerResult> {
         logger.info(`Ingested TeamMemberCreated event: ${JSON.stringify(event.data)}`);
-
-        return await SuccessPromise;
+        try {
+            this.succeedEvent();
+            return await SuccessPromise;
+        } catch (error) {
+            this.failEvent();
+            throw new Error(error);
```

Comment:
Just rethrow the error here
```suggestion
            throw error;
```